src/workerd/io/worker.c++

```diff
@@ -720,10 +720,7 @@ static void stopProfiling(jsg::Lock& js,
       }
 
       auto hitLineCount = allNodes[i]->GetHitLineCount();
-      v8::CpuProfileNode::LineTick* lineBuffer =
-          (v8::CpuProfileNode::LineTick*)malloc(
-          hitLineCount * sizeof(v8::CpuProfileNode::LineTick));
-      KJ_DEFER(free(lineBuffer));
+      v8::CpuProfileNode::LineTick lineBuffer[hitLineCount];
```

Comment:
This creates a variable-length stack array, which may be dangerous depending on how large `hitLineCount` can get. There is a macro `KJ_STACK_ARRAY` which helps with allocating small arrays on the stack and big ones on the heap. Alternatively, you could just use `kj::heapArray` here to always allocate on the heap.